contents/security/Https.md

```diff
@@ -0,0 +1,62 @@
+
+- On public free wifi, a Wifi adapter in monitor mode would be able to capture all the ongoing packets to and from the wifi access point, regardless of its destination. If the traffic is transmitted over HTTP, the data sent over wifi is in plain text and the session token and password can be stolen. One famous example would be [Firesheep](https://github.com/codebutler/firesheep), a Firefox plugin to sniff session token used in websites such as Facebook. This has made Facebook [change its default protocol from HTTP to HTTPS](https://www.facebook.com/notes/facebook-engineering/secure-browsing-by-default/10151590414803920/)
+
+- Our network packets usually travel through switches and routers around the globe to reach the destination. Any one of them, if compromised, could expose our network traffic to the sniffer. [Network tap](http://searchnetworking.techtarget.com/definition/Network-tap) is an example of a device used to sniff network traffic.
+
+- Our Internet architecture relies on [DNS](https://en.wikipedia.org/wiki/Domain_Name_System) for domain name and IP mapping and [ARP](https://en.wikipedia.org/wiki/Address_Resolution_Protocol) for MAC address and IP mapping. None of the above is built with security in mind. Common attacks such as [DNS cache poisoning](https://en.wikipedia.org/wiki/DNS_spoofing) or [ARP poisoning](https://en.wikipedia.org/wiki/ARP_spoofing)
+could redirect your traffic for monitoring.
+
+All in all, the Internet architecture that we rely on for network transmission is very vulnerable to network sniffing. If we were to use HTTP, which transmits packets in plain text, no confidentiality could be guaranteed for our web application. Therefore, we need to use HTTPS as an end to end encryption to secure our network packets.
+
+### Why HTTPS is secure
+
+As aforementioned, our network is not secure, so how could HTTPS help? HTTPS is built on top of HTTPS with the addition of [SSL](https://www.digicert.com/ssl.htm) to encrypt the plain text message. The purpose of this encryption is to make sure only client and the server could decrypt the message with required keys, and sniffer cannot decrypt packets even though they may sniff packets. 
+
+There are mainly 3 encryption algorithms used in HTTPS, namely RSA, Diffie Hellman and Elliptic Curve Algorithm. They are more thoroughly explained in [Introduction to Cryptography](../security/cryptography.md) section. These algorithms prevent sniffers from decrypting packets without knowing the keys used because the best attack algorithms known at the moment run in [sub-exponential](https://en.wikipedia.org/wiki/Time_complexity#Sub-exponential_time) time. Therefore, the attack is believed to be computationally infeasible when the keysize is large enough(e.g. 2048 bits for RSA), though it is not mathematically proven.
```

Comment:
Will be, hopefully. Plan is at least to talk about RSA to complete encryption section.